### PR TITLE
Feature/elastic shell scripts

### PR DESCRIPTION
**Pull request by ONS-Tom**
- Add shell scripts for creating an ElasticSearch index or checking if an index exists, for use by [business-index-elastic-index](https://github.com/ONSdigital/business-index-elastic-index)

Although these scripts are used from a `Jenkinsfile`, they can also be used locally (although running the `business-index-api` passing in `-DONS_BI_ES_RECREATE_INDEX=true` will achieve the same thing).

### CODE REVIEW

File: conf/scripts/create_index.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# arguments for the host, port, index name and path to JSON file which includes
+# the index definition.
+
+SCRIPT_NAME=$0
+REQUIRED_NUM_ARGS=4
+
+# Fail fast if we get any errors
+set -e
+
+function usage {
```
Comment (willis7) [inline conf/scripts/create_index.sh]: I know you took this from my example (my bad) but `function` is non-standard. Use `usage()` instead.

File: conf/scripts/create_index.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    echo "usage: ${SCRIPT_NAME} host port index_name path_to_index_json"
+    echo "  host                    elasticsearch host"
+    echo "  port                    elasticsearch port"
+    echo "  index_name              name of index to create"
+    echo "  path_to_index_json      path to json file with index definition"
+    exit 1
+}
+
+# Fail the script if we recieve an incorrect number of arguments
+if [[ $# -ne REQUIRED_NUM_ARGS ]] ; then
```
Comment (willis7) [inline conf/scripts/create_index.sh]: use `if [ $# -ne REQUIRED_NUM_ARGS ] ; then`

File: conf/scripts/create_index.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+INDEX_NAME=$3
+PATH_TO_INDEX_DEFINITION=$4
+
+# Make the curl PUT request to create the index, passing in our index definition JSON
+echo 'Creating ElasticSearch index...'
+curl --fail -XPUT "${HOST}:${PORT}/${INDEX_NAME}" -d @"${PATH_TO_INDEX_DEFINITION}"
+echo
+
+echo "Successfully created ElasticSearch index [${INDEX_NAME}] with the following settings:"
+echo "$(cat ${PATH_TO_INDEX_DEFINITION})"
```
Comment (willis7) [inline conf/scripts/create_index.sh]: The `echo` is suplerfuous. Use `cat ${PATH_TO_INDEX_DEFINITION})`

File: conf/scripts/index_exists.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    echo 'Error, you need to provide the correct number of arguments.'
+    usage
+fi
+
+HOST=$1
+PORT=$2
+INDEX_NAME=$3
+
+echo "Checking if index [${INDEX_NAME}] already exists within ElasticSearch"
+http_code=$(curl --write-out "%{http_code}\n" --silent --output /dev/null "${HOST}:${PORT}/${INDEX_NAME}")
```
Comment (willis7) [inline conf/scripts/index_exists.sh]: [SC1117](https://github.com/koalaman/shellcheck/wiki/SC1117): Backslash is literal in "\n". Prefer explicit escaping: "\\\n"

File: conf/scripts/index_exists.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+function usage {
+    echo "usage: ${SCRIPT_NAME} host port index_name"
+    echo "  host                    elasticsearch host"
+    echo "  port                    elasticsearch port"
+    echo "  index_name              name of index to check exists"
+    exit 1
+}
+
+# Fail the script if we recieve an incorrect number of arguments
+if [[ $# -ne REQUIRED_NUM_ARGS ]] ; then
```
Comment (willis7) [inline conf/scripts/index_exists.sh]: as above

File: conf/scripts/index_exists.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+HOST=$1
+PORT=$2
+INDEX_NAME=$3
+
+echo "Checking if index [${INDEX_NAME}] already exists within ElasticSearch"
+http_code=$(curl --write-out "%{http_code}\n" --silent --output /dev/null "${HOST}:${PORT}/${INDEX_NAME}")
+
+echo "HTTP Code returned from checking if the index [$INDEX_NAME] exists: $http_code"
+if [[ $http_code == '200' ]] ; then
```
Comment (willis7) [inline conf/scripts/index_exists.sh]: as above

**Comment by willis7** [comment]
Your Bash foo is improving young Padawan

**Comment by ONS-Tom** [comment]
In future, it is likely that the `index_exists` will be combined with `create_index.sh`, however for now it's nice to be able to glance at Jenkins to know at what stage a job has failed, rather than looking into the logs.

**Comment by ONS-Tom** [comment]
That format makes sense, I've added a ticket `REG-1480` for a quick refactor once the rest of the pipeline has been completed.

**Pull request merged**